frontend/packages/dev-console/src/utils/useAddActionExtensions.ts

```diff
@@ -32,3 +40,68 @@ export const useAddActionExtensions = (): [ResolvedExtension<AddAction>[], boole
 
   return [allAddActionExtensions, resolved, allAddActionsDisabled];
 };
+
+export const getEnabledSubCatalogTypes = (catalogType: string): boolean => {
+  if (window.SERVER_FLAGS.developerCatalogTypes) {
+    const developerCatalogTypes = JSON.parse(window.SERVER_FLAGS.developerCatalogTypes);
+    if (developerCatalogTypes?.state === CatalogVisibilityState.Enabled) {
+      if (developerCatalogTypes?.enabled && developerCatalogTypes?.enabled.length) {
+        return developerCatalogTypes?.enabled.includes(catalogType);
+      }
+    } else {
+      if (developerCatalogTypes?.disabled && developerCatalogTypes?.disabled.length) {
+        return !developerCatalogTypes?.disabled.includes(catalogType);
+      }
+      if (developerCatalogTypes?.disabled && developerCatalogTypes?.disabled.length === 0) {
+        return false;
+      }
+    }
+  }
+  return true;
+};
+
+export const GetAllSubCatalogsDisabled = (catalogType?: string) => {
```

Comment:
you are not passing any value to catalog type parameter while calling `GetAllSubCatalogsDisabled` anywhere

Reply:
And please use a lower case `get...` as we do anywhere else for function names.

Reply:
If we want to use this function for particular catalogType, then we can use this directly by passing catalogtype.. So I kept catalogType as input 

If I use lowercase as function name.. I was getting below error since I am calling another hook inside that function.. I did some research and the suggestion was to keep starting letter as uppercase.. any feedback on this will be helpful

```
74:32  error  React Hook "useResolvedExtensions" is called in function "getAllSubCatalogsDisabled" that is neither a React function component nor a custom React Hook function  react-hooks/rules-of-hooks
  75:5   error  React Hook "React.useCallback" is called in function "getAllSubCatalogsDisabled" that is neither a React function component nor a custom React Hook function      react-hooks/rules-of-hooks
```

Reply:
I will remove `catalogType`  from input parameter